format/spec.md

```diff
@@ -702,6 +703,21 @@ Blob metadata is a struct with the following fields:
 | _optional_ | _optional_ | **`properties`** | `map<string, string>` | Additional properties associated with the statistic. Subset of Blob properties in the Puffin file. |
 
 
+#### Partition statistics
+
+Partition statistics files are the valid files based on [Partition statistics spec](../partition-statistics-spec). Partition statistics are informational. A reader can choose to
+ignore partition statistics information. Partition statistics support is not required to read the table correctly. A table can contain
+many partition statistics files associated with different table snapshots.
+
+Partition statistics files metadata within `partition-statistics` table metadata field is a struct with the following fields:
+
+| v1 | v2 | Field name                 | Type     | Description                                                                                           |
+|----|----|----------------------------|----------|-------------------------------------------------------------------------------------------------------|
+| _required_ | _required_ | **`snapshot-id`**          | `long`   | ID of the Iceberg table's snapshot the partition statistics file is associated with.                  |
+| _required_ | _required_ | **`statistics-file-path`** | `string` | Path of the partition statistics file. See [Partition statistics spec](../partition-statistics-spec). |
+| _required_ | _required_ | **`sequence-number`**      | `long`   | Sequence number of the Iceberg table's snapshot the partition statistics was computed from.           |
```

Comment:
compaction can write to starting sequence number, would it mess up with the usage of this value?

Reply:
I might still have some gaps in understanding sequence numbers fully.

But I see that Puffin also saves it. So, I followed the same.
https://github.com/apache/iceberg/blob/master/format/spec.md#table-statistics 

Do you have any suggestions for this?

Reply:
thought about it a bit more last night, it should be fine for this use case, because the same sequence number means the data might have been replaced but the logical content remains the same, so we should be good here.

Reply:
ack

Reply:
We always assign a new sequence number, even if we are compacting and setting the data sequence number for new files. So there's some nuance here.

We could use something like "max data sequence number". That would give us an idea when these are equivalent. If a given snapshot's max data sequence number is equal to a known file, the stats should be identical.

Reply:
changed to _max-data-sequence-number_